docs/docs/ref-03-component-specs.md

```diff
@@ -37,6 +37,10 @@ object getInitialState()
 
 Invoked once before the component is mounted. The return value will be used as the initial value of `this.state`.
 
+> Note:
+>
+> This method is not available on ES6 `class` components that extend `React.Component`. It may be removed entirely in a future version of React.
```

Comment:
Could add a recommendation to enable property initializers in Babel to set initial state that way. https://babeljs.io/blog/2015/06/07/react-on-es6-plus#property-initializers

As a user if something was being taken away, I'd definitely want a suggestion on a replacement pattern like this.